sdk/servicebus/Azure.Messaging.ServiceBus/tests/RuleManager/RuleManagerLiveTests.cs

```diff
@@ -0,0 +1,588 @@
+
+                var expectedOrders = Orders.Where(c => c.Color == "yellow").ToList();
+                await ReceiveMessages(
+                    client,
+                    scope.TopicName,
+                    scope.SubscriptionNames.First(),
+                    expectedOrders);
+            }
+        }
+
+        [Test]
+        public async Task SqlFilterOnTheUserProperties()
+        {
+            await using (var scope = await ServiceBusScope.CreateWithTopic(enablePartitioning: false, enableSession: false))
+            {
+                await using var client = new ServiceBusClient(TestEnvironment.ServiceBusConnectionString);
+
+                ServiceBusRuleManager ruleManager = client.CreateRuleManager(scope.TopicName, scope.SubscriptionNames.First());
+
+                IEnumerable<RuleDescription> rulesDescription = await ruleManager.GetRulesAsync();
+                Assert.True(rulesDescription.Count() == 1);
+                Assert.AreEqual(RuleDescription.DefaultRuleName, rulesDescription.First().Name);
+
+                await ruleManager.RemoveRuleAsync(RuleDescription.DefaultRuleName);
+
+                await ruleManager.AddRuleAsync(new RuleDescription
+                {
+                    Filter = new SqlFilter("Color = 'yellow'"),
+                    Name = "SqlUserPropertyRule"
+                });
+
+                rulesDescription = await ruleManager.GetRulesAsync();
+                Assert.True(rulesDescription.Count() == 1);
+                Assert.AreEqual("SqlUserPropertyRule", rulesDescription.First().Name);
+
+                await SendMessages(client, scope.TopicName);
+
+                var expectedOrders = Orders.Where(c => c.Color == "yellow").ToList();
+                await ReceiveMessages(
+                    client,
+                    scope.TopicName,
+                    scope.SubscriptionNames.First(),
+                    expectedOrders);
+            }
+        }
+
+        [Test]
+        public async Task SqlFilterWithAction()
+        {
+            await using (var scope = await ServiceBusScope.CreateWithTopic(enablePartitioning: false, enableSession: false))
+            {
+                await using var client = new ServiceBusClient(TestEnvironment.ServiceBusConnectionString);
+
+                ServiceBusRuleManager ruleManager = client.CreateRuleManager(scope.TopicName, scope.SubscriptionNames.First());
+
+                IEnumerable<RuleDescription> rulesDescription = await ruleManager.GetRulesAsync();
+                Assert.True(rulesDescription.Count() == 1);
+                Assert.AreEqual(RuleDescription.DefaultRuleName, rulesDescription.First().Name);
+
+                await ruleManager.RemoveRuleAsync(RuleDescription.DefaultRuleName);
+
+                await ruleManager.AddRuleAsync(new RuleDescription
+                {
+                    Filter = new SqlFilter("Color = 'blue'"),
+                    Action = new SqlRuleAction("SET Priority = 'high'"),
```

Comment:
Don't we want to assert that the Priority gets set to high in the messages?

Also, is the property name case-sensitive?

Reply:
Ah..missed to assert the Priority. I thought about it but totally slipped out of my mind.

Good Catch!

Nope, property name is not case-sensitive.

Reply:
Ah..missed to assert the Priority. I thought about it but totally slipped out of my mind.

Good catch!

Nope, property name is not case-sensitive.